packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/template/example/ModelExampleWriter.java

```diff
@@ -108,16 +108,24 @@ public static class ExampleNodeAssertionVisitor {
 
         private final List<String> assertions = new ArrayList<>();
 
-        private void addEqualsAssertion(String expected, String code) {
-            assertions.add(String.format("Assertions.assertEquals(%1$s, %2$s);", expected, code));
+        private void addEqualsAssertion(String expected, String code, boolean booleanAssertion) {
+            if (booleanAssertion) {
+                if (Boolean.parseBoolean(expected)) {
+                    assertions.add(String.format("Assertions.assertTrue(%s);", code));
+                } else {
+                    assertions.add(String.format("Assertions.assertFalse(%s);", code));
+                }
+            } else {
+                assertions.add(String.format("Assertions.assertEquals(%1$s, %2$s);", expected, code));
+            }
         }
 
         public void accept(ExampleNode node, String getterCode) {
             if (node instanceof LiteralNode) {
                 node.getClientType().addImportsTo(imports, false);
 
                 addEqualsAssertion(node.getClientType().defaultValueExpression(((LiteralNode) node).getLiteralsValue()),
-                    getterCode);
+                    getterCode, node.getClientType().asNullable() == ClassType.BOOLEAN);
```

Comment:
Is this still needed?

Reply:
Yeah, guess we could make a little improvement on boolean assertions.